x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/rules_table/rules_table_facade.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import type { Dispatch } from 'react';
+import { Rule, FilterOptions, PaginationOptions } from '../types';
+import { RulesTableAction, LoadingRuleAction } from './rules_table_reducer';
+
+export interface RulesTableFacade {
+  setRules(newRules: Rule[], newPagination: Partial<PaginationOptions>): void;
+  updateRules(rules: Rule[]): void;
+  updateOptions(filter: Partial<FilterOptions>, pagination: Partial<PaginationOptions>): void;
+  actionStarted(actionType: LoadingRuleAction, ruleIds: string[]): void;
+  actionStopped(): void;
+  setShowIdleModal(show: boolean): void;
+  setLastRefreshDate(): void;
+  setAutoRefreshOn(on: boolean): void;
+}
```

Comment:
Naming as "facade" and even introduction of this guy is a temporary measure. I'd revisit this later when I start refactoring rules management logic. If you have a better naming suggestion for short term, I'd be glad to rename though.